netket/utils/types.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Tuple, Callable
+
+PRNGKey = Any
+Shape = Tuple[int]
+Dtype = Any  # this could be a real type?
+Array = Any
```

Comment:
```suggestion
from typing import Any, Sequence, Callable

PRNGKey = Any
Shape = Sequence[int]
Dtype = Any  # this could be a real type?
Array = Any
```
Since we're discussing types: JAX uses `Sequence` for `Shape`, probably we should too.